### ISSUE TITLE
Destructured `goog.module.get` calls cause errors

### ISSUE THREAD

**Issue by HarryCutts** [google/closure-compiler#2196]
Calling `goog.module.get` in a destructuring assignment statement causes a compiler error. For example:

```js
goog.require('goog.testing.jsunit');

goog.scope(function() {
  const {BASE_PATH} = goog.module.get('goog.testing.jsunit');
});
```

This code causes a `JSC_GOOG_SCOPE_NON_ALIAS_LOCAL` error (as can be seen [in the online compiler][example]):

```
JSC_GOOG_SCOPE_NON_ALIAS_LOCAL: The local variable BASE_PATH is in a goog.scope and is not an alias. at line 4 character 9
  const {BASE_PATH} = goog.module.get('goog.testing.jsunit');
```

Replacing the problematic line with the following more traditional assignment works fine:

```js
const BASE_PATH = goog.module.get('goog.testing.jsunit').BASE_PATH;
```

[example]: https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540use_closure_library%2520true%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250Agoog.require('goog.testing.jsunit')%253B%250A%250Agoog.scope(function()%2520%257B%250A%2520%2520const%2520%257BBASE_PATH%257D%2520%253D%2520goog.module.get('goog.testing.jsunit')%253B%250A%2520%2520%252F%252Fconst%2520BASE_PATH%2520%253D%2520goog.module.get('goog.testing.jsunit').BASE_PATH%253B%250A%257D)%253B

**Comment by concavelenz** [comment]
Is there are reason you can't use goog.module and avoid goog.scope altogether?

**Comment by HarryCutts** [comment]
No, there isn't, except that my team doesn't have time to do it right now.

**Comment by concavelenz** [comment]
Do what exactly? Migrating typically one unit test file is trivial.

**Comment by HarryCutts** [comment]
Well, the actual issue isn't occurring in a unit test file, I was just using `goog.testing.jsunit` as an example. But anyway, I suppose it might be quick to migrate the `require`ing module too, but surely the point of `goog.module.get` is that it's available in cases where doing that wouldn't be quick or simple.

**Comment by concavelenz** [comment]
The problem isn't goog.module.get, the problem is goog.scope and its limitations. We aren't investing in the goog.scope code.  You can do this:

  const BASE_PATH = goog.module.get('goog.testing.jsunit').BASE_PATH;  

should work.

**Comment by evmar** [comment]
I suggest retitling this bug to reflect the broader problem:

Cannot use destructuring in goog.scope

Or some such.
